activiti-services/activiti-services-rest/src/main/java/org/activiti/services/rest/controllers/TaskVariableController.java

```diff
@@ -0,0 +1,87 @@
+ * limitations under the License.
+ *
+ */
+
+package org.activiti.services.rest.controllers;
+
+import java.util.Map;
+
+import org.activiti.engine.TaskService;
+import org.activiti.services.core.ProcessEngineWrapper;
+import org.activiti.services.core.model.TaskVariables;
+import org.activiti.services.core.model.TaskVariables.TaskVariableScope;
+import org.activiti.services.core.model.commands.SetTaskVariablesCmd;
+import org.activiti.services.rest.resources.assembler.TaskVariableResourceAssembler;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.hateoas.MediaTypes;
+import org.springframework.hateoas.Resource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/v1/tasks/{taskId}/variables",
+        produces = MediaTypes.HAL_JSON_VALUE)
+public class TaskVariableController {
+
+    private ProcessEngineWrapper processEngine;
+
+    private final TaskService taskService;
+
+    private final TaskVariableResourceAssembler variableResourceBuilder;
+
+    @Autowired
+    public TaskVariableController(ProcessEngineWrapper processEngine,
+                                  TaskService taskService,
+                                  TaskVariableResourceAssembler variableResourceBuilder) {
+        this.processEngine = processEngine;
+        this.taskService = taskService;
+        this.variableResourceBuilder = variableResourceBuilder;
+    }
+
+    @RequestMapping(value = "/",
+            method = RequestMethod.GET)
+    public Resource<Map<String, Object>> getVariables(@PathVariable String taskId) {
+        Map<String, Object> variables = taskService.getVariables(taskId);
+        return variableResourceBuilder.toResource(new TaskVariables(taskId, variables, TaskVariableScope.GLOBAL));
+    }
+
+    @RequestMapping(value = "/local",
+            method = RequestMethod.GET)
+    public Resource<Map<String, Object>> getVariablesLocal(@PathVariable String taskId) {
+        Map<String, Object> variables = taskService.getVariablesLocal(taskId);
+        return variableResourceBuilder.toResource(new TaskVariables(taskId, variables, TaskVariableScope.LOCAL));
+    }
+
+    @RequestMapping(value = "/",
+            method = RequestMethod.POST)
+    public ResponseEntity<Void> setVariables(@PathVariable String taskId,
+                                             @RequestBody(required = false) SetTaskVariablesCmd setTaskVariablesCmd) {
+        processEngine.setTaskVariables(setTaskVariablesCmd);
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
+
+    @RequestMapping(value = "/local",
```

Comment:
I think that it is easier to understand if the endpoint is separated from the query parameter.

If specify the value of scope in the query parameter, I think that the developer is confused as to what to specify for the code value of scope.

Actually, in the old REST, the scope is specified by the parameter, but from the customers I often be asked what to specify for the code value.